vaadin-charts-flow-parent/vaadin-charts-flow-svg-generator/src/main/java/com/vaadin/flow/component/charts/export/NodeRunner.java

```diff
@@ -13,23 +13,54 @@
 
 package com.vaadin.flow.component.charts.export;
 
+import java.io.File;
 import java.io.IOException;
 import java.util.ArrayList;
 import java.util.List;
 
-import com.vaadin.flow.server.frontend.FrontendTools;
-import com.vaadin.flow.server.frontend.FrontendToolsSettings;
+import com.vaadin.flow.server.frontend.FrontendToolsLocator;
 import com.vaadin.flow.server.frontend.FrontendUtils;
 
 class NodeRunner {
 
+    private final FrontendToolsLocator frontendToolsLocator;
+
+    NodeRunner() {
+        this(new FrontendToolsLocator());
+    }
+
+    NodeRunner(FrontendToolsLocator frontendToolsLocator) {
+        this.frontendToolsLocator = frontendToolsLocator;
+    }
+
+    /**
+     * Returns the absolute file path to a Node.js executable.
+     * First tries to find a global installation, and as a fallback looks for an installation in the Vaadin home directory.
+     * Throws if no Node.js installation could be found.
+     * @return the absolute path to the Node.js executable
+     * @throws IllegalStateException when no Node.js installation could be found
+     */
+    String findNodeExecutable() {
+        // Try resolve global node installation
+        String nodeExecutableName = FrontendUtils.isWindows() ? "node.exe" : "node";
+        File nodeExecutableFile = frontendToolsLocator.tryLocateTool(nodeExecutableName).orElse(null);
+        if (nodeExecutableFile != null) {
+            return nodeExecutableFile.getAbsolutePath();
+        }
+        // Try resolve installation from Vaadin home
+        // This covers development setups where developers rely on the Node.js installation provided by Flow's frontend toolchain
+        nodeExecutableFile = new File(FrontendUtils.getVaadinHomeDirectory(), FrontendUtils.isWindows() ? "node/node.exe" : "node/node");
+        if (frontendToolsLocator.verifyTool(nodeExecutableFile)) {
+            return nodeExecutableFile.getAbsolutePath();
+        }
+
+        throw new IllegalStateException("Node not found");
```

Comment:
The exception could be more explicit that SVG generator requires node, and where we tried to locate it.

Reply:
Improved the message.